### PR TITLE
fix(forms): tab type for Platform forms

### LINKED ISSUE #2650: Wrong tab type when opening new Platform form
__Describe the Bug__

A newly created form is **always** attributed the type "form-cloud".

As far as I've noticed, this isn't affecting anything currently. Because there aren't any differences between the two (except for the engine select). But it might be harmful if we decide to move in different directions in the future.

__Steps to Reproduce__

1. debug `createTabForFile` 
2. create new Platform form
3. see that the type attributed is "cloud-form"

__Expected Behavior__

The type attributed should be
* "form" when opening a Platform form
* "cloud-form" when opening a Cloud form

We should also consider adding tests for this.

__Environment__

- OS: MacOS 11.6
- Camunda Modeler Version: latest
- Execution Platform: Camunda Platform

### PR DESCRIPTION
**Pull request by smbea**
Closes #2650

### CODE REVIEW

File: client/src/app/TabsProvider.js [hunk 1]
```diff
@@ -344,7 +344,13 @@ export default class TabsProvider {
         exports: {},
         extensions: [ 'form' ],
         canOpen(file) {
-          return file.name.endsWith('.form');
+          const {
+            contents
+          } = file;
+
+          const executionPlatform = JSON.parse(contents).executionPlatform;
```
Comment (barmac) [inline client/src/app/TabsProvider.js:351]: Let's make sure this does not blow up when `contents` cannot be parsed. `JSON.parse` will throw an error in that case. You can try out to pass a BPMN file here.

Comment (barmac) [reply client/src/app/TabsProvider.js:351]: Perhaps a broken JSON?

`{ myBrokenJson: true }`

**Comment by barmac** [comment]
Let's add just one test case and I think we are good to go.

**Comment by smbea** [comment]
@barmac I added a test case for when there isn't an execution platform defined in the `.form` file (following the lead of the engine select).

With form files, we always need a "type" property. So we if its not present, we get a dialog saying it's invalid. If the format is invalid (unparsable), the log shows an error message. Should I do something else here? If so, what exactly?

**Comment by barmac** [comment]
Please have a look at the capture below. The test case I mentioned is about a `.form` file which is broken in some way. We don't want the Tab Provider to break in such case but rather just inform the user that it's not a proper FORM file.

https://user-images.githubusercontent.com/28307541/147110492-f9fa2ff0-68b8-4869-a540-dc2880dc9106.mov

**Comment by smbea** [comment]
@barmac 

> Please have a look at the capture below. The test case I mentioned is about a `.form` file which is broken in some way. We don't want the Tab Provider to break in such case but rather just inform the user that it's not a proper FORM file.

This what I get if I add a `.form` file similar to what you used: 
image

**Comment by barmac** [comment]
Please try out this one: [BROKEN.txt](https://github.com/camunda/camunda-modeler/files/7763585/BROKEN.txt) (I had to change the extension because GH wouldn't allow me to upload .form file).

**Comment by smbea** [comment]
> Please try out this one: [BROKEN.txt](https://github.com/camunda/camunda-modeler/files/7763585/BROKEN.txt) (I had to change the extension because GH wouldn't allow me to upload .form file).

Ok, got it. If I just add a `try/catch`, without doing anything in the `catch` statement, then it handles the error with a dialog:

image

but should I be doing anything else inside the `catch` statement?

**Comment by smbea** [comment]
@barmac all good now

**Review by barmac — APPROVED** [review]

**Pull request merged**